apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerConnectionStringBuilderTests.cs

```diff
@@ -215,5 +215,82 @@ public void WithDatabase()
+            // DbConnectionStringBuilder lower-cases keywords, annoyingly.
+            Assert.Equal("clrtospannertypedefaultmappings=DecimalToNumeric", builder.ToString());
+            // Check that internal mapping dictionary is populated with user provided configuration.
+            Assert.True(builder.ClrToSpannerTypeMappings.Count == 1);
+            Assert.Equal(SpannerDbType.Numeric, builder.ClrToSpannerTypeMappings[typeof(decimal)]);
+
+            // Explicitly set valid type mappings.Latest should be reflected.
+            builder.ClrToSpannerTypeDefaultMappings = "DecimalToFloat64";
+            // DbConnectionStringBuilder lower-cases keywords, annoyingly.
+            Assert.Equal("clrtospannertypedefaultmappings=DecimalToFloat64", builder.ToString());
+            // Check that internal mapping dictionary is populated with user provided configuration.
+            Assert.True(builder.ClrToSpannerTypeMappings.Count == 1);
+            Assert.Equal(SpannerDbType.Float64, builder.ClrToSpannerTypeMappings[typeof(decimal)]);
+        }
+
+        [Fact]
+        public void SpannerToClrTypeDefaultMappings()
+        {
+            // Defaults - When property is not set.
+            var builder = new SpannerConnectionStringBuilder();
+            Assert.True(string.IsNullOrEmpty(builder.SpannerToClrTypeDefaultMappings));
+            // Check that internal mapping dictionary is populated with defaults.
+            Assert.True(builder.SpannerToClrTypeMappings.Count == 1);
+            Assert.Equal(typeof(double), builder.SpannerToClrTypeMappings[SpannerDbType.Float64]);
+
+            // Explicitly set valid type mappings.
+            builder = new SpannerConnectionStringBuilder("SpannerToClrTypeDefaultMappings=Float64ToSpannerNumeric");
+            // DbConnectionStringBuilder lower-cases keywords, annoyingly.
+            Assert.Equal("spannertoclrtypedefaultmappings=Float64ToSpannerNumeric", builder.ToString());
+            // Check that internal mapping dictionary is populated with user provided configuration.
+            Assert.True(builder.SpannerToClrTypeMappings.Count == 1);
+            Assert.Equal(typeof(SpannerNumeric), builder.SpannerToClrTypeMappings[SpannerDbType.Float64]);
+
+            // Explicitly set valid type mappings. Latest should be reflected.
+            builder.SpannerToClrTypeDefaultMappings = "Float64ToDecimal";
+            // DbConnectionStringBuilder lower-cases keywords, annoyingly.
+            Assert.Equal("spannertoclrtypedefaultmappings=Float64ToDecimal", builder.ToString());
+            // Check that internal mapping dictionary is populated with user provided configuration.
+            Assert.True(builder.SpannerToClrTypeMappings.Count == 1);
+            Assert.Equal(typeof(decimal), builder.SpannerToClrTypeMappings[SpannerDbType.Float64]);
+        }
+
+        [Theory]        
+        [InlineData("DecimalToFloat64,DecimalToNumeric;")] // Multiple mappings for a type.
```

Comment:
It's not clear to me from this test whether *just* additional whitespace is a problem, e.g. "DecimalToFloat64    ". Every test that includes whitespace is *also* broken in some other way.